### PR TITLE
feat: add flex grow prop to box comp

### PR DESCRIPTION
**Pull request by AlanRad-QH**
## Description

Add the missing flex prop `flex-grow` to the existing `Box` component

💁‍

## Related issues

https://app.shortcut.com/qantasaccommodation/story/114065/amend-guest-details-form

💪

## Looks like this

N/A

🤨

---

**PR check-list**

🚨 **IMPORTANT:** Please review the [guidelines for contributing](./CONTRIBUTING.md) to this repository first! 👀

- [ ] You've checked if you need to update any relevant typescript declaration file(s), for example component props (including styled-system props) or theme.
- [ ] You've requested a review from at least ONE person. Preferably a roo-ui maintainer ([Philip](https://github.com/philipwindeyer), [Eric](https://github.com/eneo5541), [Mike](https://github.com/memcc), [Marcel](https://github.com/Anthropic), [Sam](https://github.com/SamuSan), [Long](https://github.com/KieraDOG)) or a Roo UI working-group human.
  - Note: If your PR has not been reviewed within two days of requesting a review, feel free to seek reviews from other [team members of Qantas Hotels](https://github.com/orgs/hooroo/people).
- [ ] Your final merge commit message is in the format `<type>(<scope>): <description>` as per [the contributing guide](https://github.com/hooroo/roo-ui/blob/master/.github/CONTRIBUTING.md#commit-naming)
  - Example: `fix(dependencies): minor version bump to resolve security vulnerability in set-value`

### CODE REVIEW

File: src/components/Box/Box.jsx [hunk 1]
```diff
@@ -92,6 +95,7 @@ Box.propTypes = {
   ...fontWeight.propTypes,
   ...fontSize.propTypes,
   ...order.propTypes,
+  flexGrow: PropTypes.number,
```
Comment (philipwindeyer) [inline src/components/Box/Box.jsx:98]: Best also update the typedef as well, at https://github.com/hooroo/roo-ui/blob/master/src/components/index.d.ts#L135

Comment (AlanRad-QH) [reply src/components/Box/Box.jsx]: @ajlu  helped me understand how to do it like the other existing ones. Cheers mate

Comment (AlanRad-QH) [reply src/components/Box/Box.jsx]: Updated now

Comment (ajlu) [reply src/components/Box/Box.jsx]: https://github.com/hooroo/roo-ui/blob/master/src/components/index.d.ts#L240

Comment (AlanRad-QH) [reply src/components/Box/Box.jsx]: @philipwindeyer Thanks for the tip. Updated 👌

**Review by ajlu — APPROVED** [review]
🚀

**Review by yiyang-qantashotels — APPROVED** [review]

**Pull request merged**